packages/ipfs-core/src/components/pin/remote.js

```diff
@@ -0,0 +1,441 @@
+    this.swarm = swarm
+    this.service = new PinRemoteServiceAPI({ config, swarm, peerId })
+  }
+
+  /**
+   * Asks a remote pinning service to pin an IPFS object from a given path
+   *
+   * @param {string|CID} cid
+   * @param {object} options
+   * @param {string} options.service - name of a configured remote pinning service
+   * @param {?string} options.name - optional descriptive name for pin
+   * @param {?Object<string, string>} options.meta - optional metadata to attach to pin
+   * @param {?number} options.timeout - request timeout (seconds)
+   * @param {?boolean} options.background - If true, add returns remote a pin object as soon as the remote service responds.
+   * The returned pin object may have a status of 'queued' or 'pinning'.
+   * If false, the add method will not resolve until the pin status is 'pinned' before returning.
+   * When background==false and the remote service returns a status of 'failed', an Error will be thrown.
+   * @returns {Promise<RemotePin>}
+   */
+  async add (cid, options) {
+    const { service, ...addOpts } = options
+    if (!service) {
+      throw new Error('service name must be passed')
+    }
+    const svc = this.service.serviceNamed(service)
+    return svc.add(cid, addOpts)
+  }
+
+  /**
+   * List objects that are pinned by a remote service.
+   *
+   * @param {object} options
+   * @param {string} options.service - name of a configured remote pinning service
+   * @param {?Array<string|CID>} options.cid - return pins for the specified CID(s)
+   * @param {?string} options.name - return pins that contain the provided value (case-sensitive, exact match)
+   * @param {?Array<PinStatus>} options.status - return pins with the specified statuses (queued, pinning, pinned, failed). Default: pinned
+   * @param {?number} options.timeout - request timeout (seconds)
+   * @returns {AsyncGenerator<RemotePin>}
+   */
+  async * ls (options) {
+    const { service, ...lsOpts } = options
+    if (!service) {
+      throw new Error('service name must be passed')
+    }
+    const svc = this.service.serviceNamed(service)
+    for await (const res of svc.ls(lsOpts)) {
+      yield res
+    }
+  }
+
+  /**
+   * Remove a single pin from a remote pinning service.
+   * Fails if multiple pins match the specified criteria. Use rmAll to remove all pins that match.
+   *
+   * @param {object} options
+   * @param {string} options.service - name of a configured remote pinning service
+   * @param {Array<string>} options.cid - CID(s) to remove from remote pinning service
+   * @param {?Array<PinStatus>} options.status - only remove pins that have one of the specified statuses (queued, pinning, pinned, failed). Default: pinned
+   * @param {?number} options.timeout - request timeout (seconds)
+   * @returns {Promise<void>}
+   */
+  async rm (options) {
+    const { service, ...rmOpts } = options
+    if (!service) {
+      throw new Error('service name must be passed')
+    }
```

Comment:
Might be worth consolidating handling lack of `service` option under `this.service.serviceNamed` instead.